### PR TITLE
Veoss 599 new

### PR DESCRIPTION
**Pull request by oliwiaKot**
adding new pages per journey C prototype
adding check loop mode navigation

### CODE REVIEW

File: app/services/VatReturnService.scala [hunk 1]
```diff
@@ -39,6 +44,30 @@ class VatReturnService @Inject()() {
         VatReturnRequest(vrn, period, None, None, salesFromNi, salesFromEu)
     )
 
+  def getVatOwedToCountryOnReturn(country: Country, period: Period)(implicit hc: HeaderCarrier, ec: ExecutionContext) = {
+    connector.get(period).map {
+      case Right(vatReturn) =>
+        val sumFromNiToSelectedCountry = vatReturn.salesFromNi.filter(_.countryOfConsumption.code == country.code)
+          .flatMap(x => x.amounts.map(_.vatOnSales.amount)).sum
```
Comment (DevAndyroo) [inline app/services/VatReturnService.scala:51]: can we rename x variables here

File: app/services/VatReturnService.scala [hunk 2]
```diff
@@ -39,6 +44,30 @@ class VatReturnService @Inject()() {
...
+        val salesFromEU = vatReturn.salesFromEu.flatMap(_.sales)
+        val sumFromEUToSelectedCountry = salesFromEU.filter(_.countryOfConsumption.code == country.code)
+          .flatMap(x => x.amounts.map(_.vatOnSales.amount)).sum
+        val vatOwedToCountryOnPrevReturn = sumFromEUToSelectedCountry + sumFromNiToSelectedCountry
+        vatOwedToCountryOnPrevReturn
+      case Left(value) =>
+        logger.error(s"there was an error $value")
```
Comment (DevAndyroo) [inline app/services/VatReturnService.scala:58]: can we add some context to the error log as to what's gone wrong

File: app/services/corrections/CorrectionService.scala [hunk 3]
```diff
@@ -95,4 +100,16 @@ class CorrectionService @Inject()(
     }
   }
 
+  def getCorrectionsForPeriod(period: Period)(implicit ec: ExecutionContext, hc: HeaderCarrier): Future[Seq[CorrectionToCountry]] = {
+    connector.getForCorrectionPeriod(period).map{
+      response => response match {
+        case Right(payloads) => {
+          payloads.flatMap{payload => payload.corrections.flatMap{_.correctionsToCountry}}}
+        case Left(error) =>
+          logger.error(s"there was an error $error")
```
Comment (DevAndyroo) [inline app/services/corrections/CorrectionService.scala:109]: same here

File: test/connectors/corrections/CorrectionConnectorSpec.scala [hunk 4]
```diff
@@ -77,4 +77,40 @@ class CorrectionConnectorSpec extends SpecBase with WireMockHelper with EitherVa
       }
     }
   }
+
+  ".getForCorrectionPeriod" - {
```
Comment (DevAndyroo) [inline test/connectors/corrections/CorrectionConnectorSpec.scala:81]: can we add a scenario or two to cover Left response

**Review by DevAndyroo — CHANGES_REQUESTED** [review]

File: app/services/VatReturnService.scala (hunk 1 above)
Comment (oliwiaKot) [reply app/services/VatReturnService.scala]: done

File: app/services/corrections/CorrectionService.scala (hunk 3 above)
Comment (oliwiaKot) [reply app/services/corrections/CorrectionService.scala]: done now

File: test/connectors/corrections/CorrectionConnectorSpec.scala (hunk 4 above)
Comment (oliwiaKot) [reply test/connectors/corrections/CorrectionConnectorSpec.scala:81]: added 2 scenarios

File: app/services/VatReturnService.scala (hunk 2 above)
Comment (oliwiaKot) [reply app/services/VatReturnService.scala]: done now

**Review by DevAndyroo — APPROVED** [review]

**Pull request merged**